app/core/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -748,16 +748,20 @@ private void syncFolder(Account account, String folder, MessagingListener listen
         }
     }
 
-    private SyncConfig createSyncConfig(Account account) {
+    private SyncConfig createSyncConfig(Account account, boolean inRemoteSearch) {
         return new SyncConfig(
                     account.getExpungePolicy().toBackendExpungePolicy(),
-                    account.getEarliestPollDate(),
+                    (inRemoteSearch? null : account.getEarliestPollDate()),
```

Comment:
:+1:

Reply:
There are still some – however, comparably cosmetic – issues when you change flags of an old message. `inSearch` or `inRemoteSearch` should probably be accessible at all relevant places.